core/src/main/java/fr/sncf/osrd/envelope_sim/overlays/EnvelopeAcceleration.java

```diff
@@ -0,0 +1,29 @@
+package fr.sncf.osrd.envelope_sim.overlays;
+
+import fr.sncf.osrd.envelope.StepConsumer;
+import fr.sncf.osrd.envelope_sim.Action;
+import fr.sncf.osrd.envelope_sim.PhysicsPath;
+import fr.sncf.osrd.envelope_sim.PhysicsRollingStock;
+import fr.sncf.osrd.envelope_sim.TrainPhysicsIntegrator;
+
+public class EnvelopeAcceleration {
+    /** Accelerate, storing the resulting steps into consumer */
+    public static void accelerate(
+            PhysicsRollingStock rollingStock,
+            PhysicsPath path,
+            double timeStep,
+            double startPosition,
+            double startSpeed,
+            StepConsumer consumer
+    ) {
+        double position = startPosition;
+        double speed = startSpeed;
+        while (true) {
+            var step = TrainPhysicsIntegrator.step(rollingStock, path, timeStep, position, speed, Action.ACCELERATE, 1);
+            position += step.positionDelta;
+            speed = step.speed;
+            if (consumer.addStep(position, speed, step.timeDelta))
```

Comment:
addStep returns false when the step is added, that is a counter-intuitive behavior. Could we set this to true whe the step has been added and false otherwise, then this if would become
`if (!consumer.addStep(position, speed, step.timeDelta))`